robin.user.js

```diff
@@ -310,7 +324,9 @@
         var channel_needle = $("#chat-prepend-select").val().trim();
         var source = String($(".text-counter-input").val());
 
-        if(settings.filterChannel && !(source.toLowerCase().startsWith(channel_needle.toLowerCase()))) {
+        if (selectedChannel >= 0 && !(source.toLowerCase().startsWith(channelList[selectedChannel].toLowerCase())))
```

Comment:
Looking at the code, it actually is. Don't forget that Javascript has variable hoisting. So all variables are treated like they are at the top of the function/scope. So technically it is at minimum -1 based on the lines a few lines down.